File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/LineageNodeDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Lineage for a data entity.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=LineageNodeDetails.Builder.class)

public final class LineageNodeDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"id", "qualifiedName", "referenceType", "name", "type", "parentId", "properties"})
    public LineageNodeDetails(String id, String qualifiedName, ReferenceType referenceType, String name, String type, String parentId, java.util.Map<String, Object> properties) {
        super();
        this.id = id;
        this.qualifiedName = qualifiedName;
        this.referenceType = referenceType;
        this.name = name;
        this.type = type;
        this.parentId = parentId;
        this.properties = properties;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Unique identifier (GUID) of the node.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("id")
private String id;

        /**
         * Unique identifier (GUID) of the node.
         * @param id the value to set
         * @return this builder
         **/
        

public Builder id(String id) {
    this.id = id;
    return this;
}
            /**
     * Fully\u2011qualified name ({@code namespace:name}) of the node.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("qualifiedName")
private String qualifiedName;

        /**
         * Fully\u2011qualified name ({@code namespace:name}) of the node.
         * @param qualifiedName the value to set
         * @return this builder
         **/
        

public Builder qualifiedName(String qualifiedName) {
    this.qualifiedName = qualifiedName;
    return this;
}
            /**
     * referenceType of the node
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("referenceType")
private ReferenceType referenceType;

        /**
         * referenceType of the node
         * @param referenceType the value to set
         * @return this builder
         **/
        

public Builder referenceType(ReferenceType referenceType) {
    this.referenceType = referenceType;
    return this;
}
            /**
     * Human\u2011readable name of the node.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("name")
private String name;

        /**
         * Human\u2011readable name of the node.
         * @param name the value to set
         * @return this builder
         **/
        

public Builder name(String name) {
    this.name = name;
    return this;
}
            /**
     * High\u2011level categorisation (e.g., WorkspaceObject, Dataset, etc.).
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("type")
private String type;

        /**
         * High\u2011level categorisation (e.g., WorkspaceObject, Dataset, etc.).
         * @param type the value to set
         * @return this builder
         **/
        

public Builder type(String type) {
    this.type = type;
    return this;
}
            /**
     * Parent of the node
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("parentId")
private String parentId;

        /**
         * Parent of the node
         * @param parentId the value to set
         * @return this builder
         **/
        

public Builder parentId(String parentId) {
    this.parentId = parentId;
    return this;
}
            /**
     * A generic property bag associated with the Node
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("properties")
private java.util.Map<String, Object> properties;

        /**
         * A generic property bag associated with the Node
         * @param properties the value to set
         * @return this builder
         **/
        

public Builder properties(java.util.Map<String, Object> properties) {
    this.properties = properties;
    return this;
}


        public LineageNodeDetails build() {
            LineageNodeDetails model = new LineageNodeDetails(this.id
                , this.qualifiedName
                , this.referenceType
                , this.name
                , this.type
                , this.parentId
                , this.properties);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(LineageNodeDetails model) {
                this.id(model.getId());
    this.qualifiedName(model.getQualifiedName());
    this.referenceType(model.getReferenceType());
    this.name(model.getName());
    this.type(model.getType());
    this.parentId(model.getParentId());
    this.properties(model.getProperties());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Unique identifier (GUID) of the node.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("id")
    private final String id;

        /**
     * Unique identifier (GUID) of the node.
     * @return the value
     **/
    
    public String getId() {
        return id;
    }


        /**
     * Fully\u2011qualified name ({@code namespace:name}) of the node.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("qualifiedName")
    private final String qualifiedName;

        /**
     * Fully\u2011qualified name ({@code namespace:name}) of the node.
     * @return the value
     **/
    
    public String getQualifiedName() {
        return qualifiedName;
    }

    
        /**
     * referenceType of the node
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("referenceType")
    private final ReferenceType referenceType;

        /**
     * referenceType of the node
     * @return the value
     **/
    
    public ReferenceType getReferenceType() {
        return referenceType;
    }


        /**
     * Human\u2011readable name of the node.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("name")
    private final String name;

        /**
     * Human\u2011readable name of the node.
     * @return the value
     **/
    
    public String getName() {
        return name;
    }


        /**
     * High\u2011level categorisation (e.g., WorkspaceObject, Dataset, etc.).
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("type")
    private final String type;

        /**
     * High\u2011level categorisation (e.g., WorkspaceObject, Dataset, etc.).
     * @return the value
     **/
    
    public String getType() {
        return type;
    }


        /**
     * Parent of the node
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("parentId")
    private final String parentId;

        /**
     * Parent of the node
     * @return the value
     **/
    
    public String getParentId() {
        return parentId;
    }


        /**
     * A generic property bag associated with the Node
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("properties")
    private final java.util.Map<String, Object> properties;

        /**
     * A generic property bag associated with the Node
     * @return the value
     **/
    
    public java.util.Map<String, Object> getProperties() {
        return properties;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("LineageNodeDetails(");
        sb.append("id=").append(String.valueOf(this.id));
        sb.append(", qualifiedName=").append(String.valueOf(this.qualifiedName));
        sb.append(", referenceType=").append(String.valueOf(this.referenceType));
        sb.append(", name=").append(String.valueOf(this.name));
        sb.append(", type=").append(String.valueOf(this.type));
        sb.append(", parentId=").append(String.valueOf(this.parentId));
        sb.append(", properties=").append(String.valueOf(this.properties));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof LineageNodeDetails)) {
            return false;
        }

        LineageNodeDetails other = (LineageNodeDetails) o;
        return java.util.Objects.equals(this.id, other.id) &&
            java.util.Objects.equals(this.qualifiedName, other.qualifiedName) &&
            java.util.Objects.equals(this.referenceType, other.referenceType) &&
            java.util.Objects.equals(this.name, other.name) &&
            java.util.Objects.equals(this.type, other.type) &&
            java.util.Objects.equals(this.parentId, other.parentId) &&
            java.util.Objects.equals(this.properties, other.properties);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.id == null ? 43 : this.id.hashCode());
        result = (result * PRIME) + (this.qualifiedName == null ? 43 : this.qualifiedName.hashCode());
        result = (result * PRIME) + (this.referenceType == null ? 43 : this.referenceType.hashCode());
        result = (result * PRIME) + (this.name == null ? 43 : this.name.hashCode());
        result = (result * PRIME) + (this.type == null ? 43 : this.type.hashCode());
        result = (result * PRIME) + (this.parentId == null ? 43 : this.parentId.hashCode());
        result = (result * PRIME) + (this.properties == null ? 43 : this.properties.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/lineage_node_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class LineageNodeDetails(object):
    """
    Lineage for a data entity.
    """

    #: A constant which can be used with the reference_type property of a LineageNodeDetails.
    #: This constant has a value of "CHILD"
    REFERENCE_TYPE_CHILD = "CHILD"

    #: A constant which can be used with the reference_type property of a LineageNodeDetails.
    #: This constant has a value of "PARENT"
    REFERENCE_TYPE_PARENT = "PARENT"

    def __init__(self, **kwargs):
        """
        Initializes a new LineageNodeDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param id:
            The value to assign to the id property of this LineageNodeDetails.
        :type id: str

        :param qualified_name:
            The value to assign to the qualified_name property of this LineageNodeDetails.
        :type qualified_name: str

        :param reference_type:
            The value to assign to the reference_type property of this LineageNodeDetails.
            Allowed values for this property are: "CHILD", "PARENT"
        :type reference_type: str

        :param name:
            The value to assign to the name property of this LineageNodeDetails.
        :type name: str

        :param type:
            The value to assign to the type property of this LineageNodeDetails.
        :type type: str

        :param parent_id:
            The value to assign to the parent_id property of this LineageNodeDetails.
        :type parent_id: str

        :param properties:
            The value to assign to the properties property of this LineageNodeDetails.
        :type properties: dict(str, object)

        """
        self.swagger_types = {
            'id': 'str',
            'qualified_name': 'str',
            'reference_type': 'str',
            'name': 'str',
            'type': 'str',
            'parent_id': 'str',
            'properties': 'dict(str, object)'
        }

        self.attribute_map = {
            'id': 'id',
            'qualified_name': 'qualifiedName',
            'reference_type': 'referenceType',
            'name': 'name',
            'type': 'type',
            'parent_id': 'parentId',
            'properties': 'properties'
        }

        self._id = None
        self._qualified_name = None
        self._reference_type = None
        self._name = None
        self._type = None
        self._parent_id = None
        self._properties = None

    @property
    def id(self):
        """
        **[Required]** Gets the id of this LineageNodeDetails.
        Unique identifier (GUID) of the node.


        :return: The id of this LineageNodeDetails.
        :rtype: str
        """
        return self._id

    @id.setter
    def id(self, id):
        """
        Sets the id of this LineageNodeDetails.
        Unique identifier (GUID) of the node.


        :param id: The id of this LineageNodeDetails.
        :type: str
        """
        self._id = id

    @property
    def qualified_name(self):
        """
        **[Required]** Gets the qualified_name of this LineageNodeDetails.
        Fully\u2011qualified name (`namespace:name`) of the node.


        :return: The qualified_name of this LineageNodeDetails.
        :rtype: str
        """
        return self._qualified_name

    @qualified_name.setter
    def qualified_name(self, qualified_name):
        """
        Sets the qualified_name of this LineageNodeDetails.
        Fully\u2011qualified name (`namespace:name`) of the node.


        :param qualified_name: The qualified_name of this LineageNodeDetails.
        :type: str
        """
        self._qualified_name = qualified_name

    @property
    def reference_type(self):
        """
        **[Required]** Gets the reference_type of this LineageNodeDetails.
        referenceType of the node

        Allowed values for this property are: "CHILD", "PARENT"


        :return: The reference_type of this LineageNodeDetails.
        :rtype: str
        """
        return self._reference_type

    @reference_type.setter
    def reference_type(self, reference_type):
        """
        Sets the reference_type of this LineageNodeDetails.
        referenceType of the node


        :param reference_type: The reference_type of this LineageNodeDetails.
        :type: str
        """
        allowed_values = ["CHILD", "PARENT"]
        if not value_allowed_none_or_none_sentinel(reference_type, allowed_values):
            raise ValueError(
                "Invalid value for `reference_type`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._reference_type = reference_type

    @property
    def name(self):
        """
        **[Required]** Gets the name of this LineageNodeDetails.
        Human\u2011readable name of the node.


        :return: The name of this LineageNodeDetails.
        :rtype: str
        """
        return self._name

    @name.setter
    def name(self, name):
        """
        Sets the name of this LineageNodeDetails.
        Human\u2011readable name of the node.


        :param name: The name of this LineageNodeDetails.
        :type: str
        """
        self._name = name

    @property
    def type(self):
        """
        **[Required]** Gets the type of this LineageNodeDetails.
        High\u2011level categorisation (e.g., WorkspaceObject, Dataset, etc.).


        :return: The type of this LineageNodeDetails.
        :rtype: str
        """
        return self._type

    @type.setter
    def type(self, type):
        """
        Sets the type of this LineageNodeDetails.
        High\u2011level categorisation (e.g., WorkspaceObject, Dataset, etc.).


        :param type: The type of this LineageNodeDetails.
        :type: str
        """
        self._type = type

    @property
    def parent_id(self):
        """
        **[Required]** Gets the parent_id of this LineageNodeDetails.
        Parent of the node


        :return: The parent_id of this LineageNodeDetails.
        :rtype: str
        """
        return self._parent_id

    @parent_id.setter
    def parent_id(self, parent_id):
        """
        Sets the parent_id of this LineageNodeDetails.
        Parent of the node


        :param parent_id: The parent_id of this LineageNodeDetails.
        :type: str
        """
        self._parent_id = parent_id

    @property
    def properties(self):
        """
        **[Required]** Gets the properties of this LineageNodeDetails.
        A generic property bag associated with the Node


        :return: The properties of this LineageNodeDetails.
        :rtype: dict(str, object)
        """
        return self._properties

    @properties.setter
    def properties(self, properties):
        """
        Sets the properties of this LineageNodeDetails.
        A generic property bag associated with the Node


        :param properties: The properties of this LineageNodeDetails.
        :type: dict(str, object)
        """
        self._properties = properties

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
